src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,241 @@
+        # torch.compile is supported starting from PyTorch 2.0
+        return False
+    if "DISABLE_TORCH_COMPILE" in config["parameters"]:
+        val = config["parameters"]["DISABLE_TORCH_COMPILE"]["string_value"].upper()
+        if val == "YES" or val == "TRUE" or val == "1":
+            return False
+    return True
+
+
+class TritonPythonModel:
+    """Your Python model must use the same class name. Every Python model
+    that is created must have "TritonPythonModel" as the class name.
+    """
+
+    def initialize(self, args):
+        """`initialize` is called only once when the model is being loaded.
+        Implementing `initialize` function is optional. This function allows
+        the model to initialize any state associated with this model.
+
+        Parameters
+        ----------
+        args : dict
+          Both keys and values are strings. The dictionary keys and values are:
+          * model_config: A JSON string containing the model configuration
+          * model_instance_kind: A string containing model instance kind
+          * model_instance_device_id: A string containing model instance device ID
+          * model_repository: Model repository path
+          * model_version: Model version
+          * model_name: Model name
+        """
+        self._model_name = args["model_name"]
+        self._logger = pb_utils.Logger
+        self._logger.log_info(
+            "Initializing model instance for '" + self._model_name + "'"
+        )
+
+        self._model_config = json.loads(args["model_config"])
+        self._inputs = _parse_io_config(self._model_config["input"])
+        self._outputs = _parse_io_config(self._model_config["output"])
+
+        kind = args["model_instance_kind"]
+        device_id = args["model_instance_device_id"]
+        self._device = torch.device(_get_device_name(kind, device_id))
+
+        model_path = _get_model_path(self._model_config)
+        if not _is_py_class_model(model_path):
+            self._logger.log_info("Loading '" + self._model_name + "' as TorchScript")
+            self._model = torch.jit.load(model_path)
+            self._model.to(self._device)
+            self._model.eval()
+            return
+
+        self._model_module = _import_module_from_path(self._model_name, model_path)
+        self._model_class = _get_model_class_from_module(self._model_module)
+        self._raw_model = self._model_class()
+        data_path = _get_model_data_path(model_path)
+        if data_path != "":
+            self._raw_model.load_state_dict(
+                torch.load(data_path, map_location=self._device)
+            )
+        else:
+            self._logger.log_info(
+                "Model parameter file not found for '" + self._model_name + "'"
+            )
+        self._raw_model.to(self._device)
+        self._raw_model.eval()
+        if _enable_torch_compile(self._model_config):
+            self._model = torch.compile(self._raw_model)
```

Comment:
torch.compile has several arguments that can be provided. See here for the documentation: https://pytorch.org/docs/stable/generated/torch.compile.html#torch.compile
We should expose these via config parameters.

Reply:
Yes, added ability to supply all optional `torch.compile` parameters. Since one of the parameter is to disable `torch.compile` (no-op as described [on the doc](https://pytorch.org/docs/stable/generated/torch.compile.html#torch.compile)), the `DISABLE_TORCH_COMPILE` option is removed as part of this change to reduce code complexity and eliminate redundancy.